domains/authorization/Proxy.IntegrationTests/Setup/ProxyWebApplicationFactory.cs

```diff
@@ -0,0 +1,134 @@
+            AuthenticationScheme.B2CAuthenticationScheme,
+            AuthenticationScheme.B2CAuthenticationScheme,
+            typeof(ProxyTestAuthHandler));
+        authenticationSchemeProvider.AddScheme(b2CScheme);
+
+        var b2CMitIdScheme = new Microsoft.AspNetCore.Authentication.AuthenticationScheme(
+            AuthenticationScheme.B2CMitIDCustomPolicyAuthenticationScheme,
+            AuthenticationScheme.B2CMitIDCustomPolicyAuthenticationScheme,
+            typeof(ProxyTestAuthHandler));
+        authenticationSchemeProvider.AddScheme(b2CMitIdScheme);
+
+        var b2CClientCredentialsScheme = new Microsoft.AspNetCore.Authentication.AuthenticationScheme(
+            AuthenticationScheme.B2CClientCredentialsCustomPolicyAuthenticationScheme,
+            AuthenticationScheme.B2CClientCredentialsCustomPolicyAuthenticationScheme,
+            typeof(ProxyTestAuthHandler));
+        authenticationSchemeProvider.AddScheme(b2CClientCredentialsScheme);
+    }
+
+    public HttpClient CreateUnauthenticatedClient()
+    {
+        var client = CreateClient();
+        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20240103");
+        return client;
+    }
+
+    public HttpClient CreateAuthenticatedClient(string sub = "", string name = "", List<string>? orgIds = null, string subType = "")
+    {
+        sub = string.IsNullOrEmpty(sub) ? Guid.NewGuid().ToString() : sub;
+        name = string.IsNullOrEmpty(name) ? "Test Testesen" : name;
+        subType = string.IsNullOrEmpty(subType) ? "user" : subType;
+        orgIds ??= new List<string> { Guid.NewGuid().ToString() };
+
+        var client = CreateClient();
+        var token = GenerateToken(sub, name, orgIds, subType);
+        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        return client;
+    }
+
+    private string GenerateToken(string sub, string name = "Default Name", List<string>? orgIds = null, string subType = "Default SubType")
+    {
+        using RSA rsa = RSA.Create(2048);
+        var req = new CertificateRequest("cn=eotest", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+        req.CertificateExtensions.Add(new X509KeyUsageExtension(X509KeyUsageFlags.DigitalSignature, true));
+        var cert = req.CreateSelfSigned(DateTimeOffset.Now, DateTimeOffset.Now.AddYears(5));
+
+        var signingCredentials = new SigningCredentials(new X509SecurityKey(cert), SecurityAlgorithms.RsaSha256);
+        var tokenHandler = new JwtSecurityTokenHandler();
+
+        var orgIdsString = string.Join(" ", orgIds ?? new List<string>());
+
+        var identity = new ClaimsIdentity(new List<Claim>
+        {
+            new("sub", sub),
+            new("name", name),
+            new("org_ids", orgIdsString),
+            new("sub_type", subType),
+        });
+
+        var securityTokenDescriptor = new SecurityTokenDescriptor
+        {
```

Comment:
Does the audience and uri need to be hard coded? Can't they be loaded in via appsetting variables?

Reply:
This is integration test. We don't have appsettings or different enviroments.